### DISCUSSION TITLE
FindService addition, "team"

### DISCUSSION THREAD

**Discussion by SenorRagequit** [exorstudios/riftbreaker-tools#145] [Wishlist] [answered]
**Wish**
---
A way to search for team in range using the FindService. 
Currently, we only got `FindEntitiesByTeamInBox` which is _not optimal_ in most scenarios.
So great addition would by to get the other ones for team too.

- FindEntitiesByTeam
- FindEntitiesByTeamInRadius
- FindEntityByTeamInDistance
- FindEntityByTeamInDistanceFromPos
- FindEntityByTeamInDistanceRange

**Why**
---
Offers an easy way to filter for all enemies and not single groups like "ground_unit" etc. without involving neutral / ambient units.

**Comment by lukaasm** [comment, marked as answer]
I don't think we will expand further native services in that way since with access to reflection we can do most of the things directly in LUA.

We should create LUA services where we could push services doing things like that:

```
function FindEntitiesByTeamInRadius( origin, radius, team )
    local predicate = 
    {
        signature = "TeamComponent",
        filter = function( entity ) 
            return EntityService:GetTeam( entity ) == team 
         end
    }

    return FindService:FindEntitiesByPredicateInRadius( origin, radius, predicate  );
end
```

**Reply by SenorRagequit** [reply]
Even though I accept this as an answer, it's a whole other level of coding.

Using services makes okay-ish coding people able to use code way easier.
Looking at the code you posted only brings up more questions.
Where does the signature come from? What is a filter? 
I understand that reflections make it **possible** to create stuff on your own, but it certainly **doesn't** make it **easy**.
It's like saying, "Of course you could just buy a car if you need one. But you have every part needed to build your own, so make a car yourself". If person XYZ has the ability to do so, they will. But how big is the % of people who know how to do that?

Just to keep in mind that not everyone is a great coder and additionally, knows how the code for the game behaves / how it all works.
